clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -157,6 +157,23 @@ def CIR_FP16 : CIR_FloatType<"FP16", "f16"> {
   }];
 }
 
+def CIR_StorageOnlyFP16 : CIR_FloatType<"StorageOnlyFP16", "f16.storage"> {
+  let summary = "CIR type that represents a storage-only fp16 type";
+  let description = [{
+    Floating-point type that represents a storage-only fp16 type.
```

Comment:
Should we call this `fp16.soft`? This comes from the usual semantics we get for more regular floats when `-msoft-float` is used, which seems to be similar here.